server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/utils/ErrorResponder.java

```diff
@@ -86,15 +92,33 @@ public HaltException haltError() {
         Preconditions.checkNotNull(type, "type must not be null in case of error");
         Preconditions.checkNotNull(message, "message must not be null in case of error");
         try {
-            return halt(statusCode, new JsonTransformer().render(new ErrorDetail(statusCode,
-                type.getType(),
-                message,
-                cause.map(e -> Optional.ofNullable(e.getMessage())).orElse(Optional.empty()))));
+            return halt(statusCode, generateBody());
         } catch (JsonProcessingException e) {
             return halt(statusCode);
         }
     }
 
+    public String asString() {
+        Preconditions.checkNotNull(statusCode, "statusCode must not be null in case of error");
```

Comment:
what happens when building the error message for displaying a good error payload throws an exception ?
We should probably be very resilient in that code path, don't you think ?

Reply:
Thanks, I think I should remove Preconditions checks